dsl/pace/dsl/stencil.py

```diff
@@ -202,6 +202,52 @@ def report_diff(arg: np.ndarray, numpy_arg: np.ndarray, label) -> str:
     return report
 
 
+@dataclasses.dataclass
+class TimingCollector:
+    build_info: Dict[str, dict] = dataclasses.field(default_factory=dict)
+    exec_info: Dict[str, dict] = dataclasses.field(default_factory=dict)
```

Comment:
Please add a docstring for the class, with an `Attributes:` section that describes what the keys and values of these dictionaries are. You can use something brief for the values (e.g. "a gt4py build_info dictionary") but it would help a lot to know if these keys are stencil names, keys of build_info, or something completely different.